vpr/src/device/rr_graph_builder.h

```diff
@@ -0,0 +1,63 @@
+#ifndef RR_GRAPH_BUILDER_H
+#define RR_GRAPH_BUILDER_H
+
+#include "rr_graph_storage.h"
+#include "rr_spatial_lookup.h"
+
+/* A data structure allows data modification on a routing resource graph 
+ *
+ * Note that the builder does not own the storage
+ * It serves a virtual protocol for
+ * - node_storage: store the node list 
+ * - node_lookup: store a fast look-up for the nodes
+ *
+ * Note:
+ * - This is the only data structre allowed to modify a routing resource graph
+ *
+ */
+class RRGraphBuilder {
+    /****************
+     * Constructors
+     ****************/
```

Comment:
... Still not a fan of these massive comments. They don't really serve a purpose (as for a well-structured class it is immediately clear where is what), so they are mostly just [counter the goal to reduce vertical whitespace](https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace). We have only so much vertical space on our monitors, so we should try to maximize information.

If it helps readability I  sometimes do that with a single line comment that looks a bit emphasized, maybe with dashes:
```
  // --- Constructors
```
Sounds like these large comments were decided before me, but I'd strongly encourage to re-evaluate :)

Reply:
Agree. These dummy lines should be removed.